File: Samples/Verbum/displaying-bookmark-names-in-the-bookmarks-pane-of-pdf-viewer-for-a-document-exported-to-p-t420469-16.1.4-/CS/BookmarkTitleUpdater/BookmarkTitleUpdater.cs
using DevExpress.XtraRichEdit.Services;

namespace BookmarkTitleUpdater
{
    public class BookmarkTitleUpdater : ILinkUpdater
    {
        public string ResolveBookmarkTitle(string bookmarkName, string currentTitle)
        {
            return bookmarkName;
        }

        public string UpdateLinkToPage(string uri, string anchor)
        {
            return anchor;
        }

        public string UpdateLinkToUri(string uri)
        {
            return uri;
        }
    }
}

File: Samples/Verbum/displaying-bookmark-names-in-the-bookmarks-pane-of-pdf-viewer-for-a-document-exported-to-p-t420469-16.1.4-/VB/BookmarkTitleUpdater/BookmarkTitleUpdater.vb
Imports DevExpress.XtraRichEdit.Services

Namespace BookmarkTitleUpdater
    Public Class BookmarkTitleUpdater
        Implements ILinkUpdater

        Public Function ResolveBookmarkTitle(ByVal bookmarkName As String, ByVal currentTitle As String) As String Implements ILinkUpdater.ResolveBookmarkTitle
            Return bookmarkName
        End Function

        Public Function UpdateLinkToPage(ByVal uri As String, ByVal anchor As String) As String Implements ILinkUpdater.UpdateLinkToPage
            Return anchor
        End Function

        Public Function UpdateLinkToUri(ByVal uri As String) As String Implements ILinkUpdater.UpdateLinkToUri
            Return uri
        End Function
    End Class
End Namespace
